tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -0,0 +1,457 @@
+            fmt_fields: self.fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Do not emit timestamps with log messages.
+    pub fn without_time(self) -> Builder<S, N, format::Format<L, ()>, W> {
+        Builder {
+            fmt_event: self.fmt_event.without_time(),
+            fmt_fields: self.fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Enable ANSI encoding for formatted events.
+    #[cfg(feature = "ansi")]
+    pub fn with_ansi(self, ansi: bool) -> Builder<S, N, format::Format<L, T>, W> {
+        Builder {
+            fmt_event: self.fmt_event.with_ansi(ansi),
+            fmt_fields: self.fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Sets whether or not an event's target is displayed.
+    pub fn with_target(self, display_target: bool) -> Builder<S, N, format::Format<L, T>, W> {
+        Builder {
+            fmt_event: self.fmt_event.with_target(display_target),
+            fmt_fields: self.fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Sets the subscriber being built to use a less verbose formatter.
+    ///
+    /// See [`format::Compact`].
+    pub fn compact(self) -> Builder<S, N, format::Format<format::Compact, T>, W>
+    where
+        N: for<'writer> FormatFields<'writer> + 'static,
+    {
+        Builder {
+            fmt_event: self.fmt_event.compact(),
+            fmt_fields: self.fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Sets the subscriber being built to use a JSON formatter.
+    ///
+    /// See [`format::Json`]
+    #[cfg(feature = "json")]
+    pub fn json(self) -> Builder<S, format::JsonFields, format::Format<format::Json, T>, W> {
+        Builder {
+            fmt_event: self.fmt_event.json(),
+            fmt_fields: format::JsonFields::new(),
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S, N, E, W> Builder<S, N, E, W> {
+    /// Sets the Visitor that the subscriber being built will use to record
+    /// fields.
+    pub fn fmt_fields<N2>(self, fmt_fields: N2) -> Builder<S, N2, E, W>
+    where
+        N2: for<'writer> FormatFields<'writer> + 'static,
+    {
+        Builder {
+            fmt_event: self.fmt_event,
+            fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S, N, E, W> Builder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Builds a [FmtLayer] infalliably.
+    pub fn finish(self) -> FmtLayer<S, N, E, W> {
```

Comment:
Should this be called `build`?

Reply:
the existing `FmtSubscriber` builder has a `finish` method, so this was named for consistency; perhaps they should both be `build`? n.b. that this would require doing a deprecation cycle for `SubscriberBuilder::finish`.

Reply:
I think that'd be worthwhile doing. Or pick some more semantically appropriate name if we can think of one.